pkgs/applications/networking/StevenBlack-hosts/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenvNoCC, fetchFromGitHub, python3, lib
+, flags ? {} ,whitelist ? ""}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "StevenBlack-hosts";
+  version = "2.5.52";
+  src = fetchFromGitHub {
+    owner = "StevenBlack";
+    repo = "hosts";
+    rev = version;
+    sha256 ="zQMzXci1/21PCvi8AGqyDataQl/kQJJ+jszxXd2XMQc=";
+  };
+
+  inherit whitelist;
+  passAsFile = ["whitelist"];
+  buildPhase =
+    let 
+      py = python3.withPackages (ps: with ps;[
+        lxml beautifulsoup4
+      ]);
+    in ''
+      ln -s $whitelistPath whitelist
+      ${py}/bin/python updateHostsFile.py \
+        ${lib.cli.toGNUCommandLineShell 
+          {} (flags // {
+            auto = true;
+            noupdate = true;
+            out = ".";
+          })
+        }
+    '';
+  
+  installPhase = ''
+    cp hosts $out
```

Comment:
It might be a good idea to not put this in `$out` directly, but rather `$out/share/StevenBlack-hosts/hosts`. This then also allows people to install the package with `environment.systemPackages` or so, after which the hosts file will be in `/run/current-system/sw/share/StevenBlack-hosts/hosts`